### ISSUE TITLE
adding reference to package .dll causes reference to dependent dll to be ignored and vice versa

### ISSUE THREAD

**Issue by sam-wheat** [NuGet/Home#5441]
Regarding .nuspec below - Looks like nuget will add project references to third party dlls only when reference to MyPackage.dll is omitted .   When the reference to MyPackage.dll is included, references to third party dlls are not created.

However, consumers of my package need references to all dlls. Is this possible?  

The file names are spelled correctly.  

All files exist in the packages lib folder when the package is installed in the consuming project. 

From [this link](https://docs.microsoft.com/en-us/nuget/schema/nuspec#specifying-explicit-assembly-references):

From the references section:

>>  If omitted, all references in lib are included. See Specifying explicit assembly references below.

This does not occur.

From the "Explicit references" section:

>>When this element is present, NuGet will add references to only the listed assemblies; 

This does not happen either.

Using Visual Studio 2017 26430.13
Nuget Package Manager 4.2.0

Thanks

```
<?xml version="1.0"?>
<package >
  <metadata>
    // 
	<references>
		<reference file="MyPackage.dll"></reference>
		<reference file="ThirdParty.dll"></reference>
		<reference file="OtherThirdParty.dll"></reference>
	</references>
  </metadata>
  
  <files>
	<file src="bin\$configuration$\*.dll" target="lib" />
  </files>
</package>

**Comment by mishra14** [comment]
@rohit21agrawal could you please take a peek?

**Comment by rohit21agrawal** [comment]
@sam-wheat is there anything special about "MyPackage.dll" ?

**Comment by leaderanalytics** [comment]
@rohit21agrawal no (I am Sam)

**Comment by rohit21agrawal** [comment]
You will have to attach a repro project for me because i cant repro this on my end

**Comment by sam-wheat** [comment]
@rohit21agrawal Ok... I created new dummy visual studio .sln and when I install the package it works fine.  However when I try to install the package to any project in my work .sln, the references are not created.  I have tried clearing the nuget cache and completely uninstalling the package.  I've tried on two different physical machines and it does not work.  What else can I try?  I can repeat this behavior at will with my work .sln.  Unfortunately I cannot send you my work .sln as it is rather large.
Thanks,

**Comment by leaderanalytics** [comment]
@rohit21agrawal 
I found it - here are steps to reproduce:

Create a new class library project.  Make sure Target Framework is .net 4.7
Create a lib folder in same folder as .csproj.
Copy any .dll to lib folder i.e. ThirdParty.dll 
Add a reference to the .dll*
Build
run C:\"Program Files (x86)"\NuGet\NuGet.exe pack MyPkg.csproj -Outputdirectory bin\release -properties Configuration=Release

Create a project to use the package:

File -> new WPF App (.Net Framework) 
Change framework to 4.7
Add package created above
Observe that the reference to ThirdParty.dll is broken


	<?xml version="1.0"?>
	<package >
	  <metadata>
		<id>MyPkg</id>
		<version>$version$</version>
		<title>Nuget reference test</title>
		<authors>Sam Wheat</authors>
		<owners>Sam Wheat</owners>
		<requireLicenseAcceptance>false</requireLicenseAcceptance>
		<description>test dummy</description>
		<releaseNotes></releaseNotes>
		<copyright>Copyright 2017</copyright>
		<tags></tags>
		<references>
			<reference file="ThirdParty.dll"></reference>
			<reference file="MyPkg.dll"></reference>
		</references>
	  </metadata>
	  
	  <files>
		<file src="bin\$configuration$\*.dll" target="lib\net47" />
	  </files>
	</package>


*Edit

**Comment by sam-wheat** [comment]
Any resolution?

Thanks

**Comment by sam-wheat** [comment]
@rohit21agrawal Please comment on steps to reproduce I have provided.  Thanks

**Comment by rohit21agrawal** [comment]
let me get back to you tomorrow on this.

**Comment by sam-wheat** [comment]
@rohit21agrawal thanks looking forward to resolving this.

**Comment by rohit21agrawal** [comment]
[issue5441consumepkg.zip](https://github.com/NuGet/Home/files/1129578/issue5441consumepkg.zip)
[issue5441.zip](https://github.com/NuGet/Home/files/1129579/issue5441.zip)


@sam-wheat couldn't repro using your steps. I have attached my projects. issue5441 is the project i used to create the nupkg (you can find the created nupkg inside the zip too.

issue5441consumepkg is the project i used to add the nuget package reference, the reference to classlibrary1 was added correctly in my case.

**Comment by sam-wheat** [comment]
@rohit21agrawal I am looking at your projects.  What version of Nuget are you using?  Also, What parameters did you use to generate the pkg?  Thanks

**Comment by rohit21agrawal** [comment]
4.3.0.4258 version of NuGet.

I am using a post 15.3 preview3 build of Visual Studio

**Comment by sam-wheat** [comment]
@rohit21agrawal  Thanks very much for your help :) 
I finally tracked this down to an old version of nugget.exe on my laptop.
For the record for anyone else:  this is a bug in nuget 3.5.  This  has been fixed in 4.1.

**Comment by MeTitus** [comment]
This is still an ongoing issue. I am running version 4.4.1 and what was working before, isn't working anymore. Files are added to the package but no references are added. I really think it is time to re-think the whole model because lately nuget.exe is all but a big mess.
